Code/Framework/AzCore/AzCore/DOM/DomUtils.cpp

```diff
@@ -59,6 +60,56 @@ namespace AZ::Dom::Utils
         return AZ::Success(AZStd::move(value));
     }
 
+    Value TypeIdToDomValue(const AZ::TypeId& typeId)
+    {
+        rapidjson::Document buffer;
+        JsonSerialization::StoreTypeId(buffer, buffer.GetAllocator(), typeId);
+        AZ_Assert(buffer.IsString(), "TypeId should be stored as a string");
+        return Value(AZStd::string_view(buffer.GetString(), buffer.GetStringLength()), true);
+    }
+
+    AZ::TypeId DomValueToTypeId(const AZ::Dom::Value& value, const AZ::TypeId* baseClassId)
+    {
+        AZ::TypeId result = AZ::TypeId::CreateNull();
+        if (value.IsString())
+        {
+            rapidjson::Value buffer;
+            buffer.SetString(value.GetString().data(), aznumeric_cast<rapidjson::SizeType>(value.GetStringLength()));
```

Comment:
Nitpick: `aznumeric_caster` will automatically detect the target type. Little bit less code to write.